app/controllers/application_controller/miq_request_methods.rb

```diff
@@ -716,7 +716,7 @@ def prov_get_form_vars
           end
         else
           field[:values].each do |v|
-            if v.class.name == "MiqHashStruct" && v.evm_object_class == :Storage
+            if v.try(:evm_object_class) == :Storage
```

Comment:
since the code is now calling `v.try(:evm_object_class)` in each leg of the complex `if` statement, would it make sense to do the call once at the top like `evm_object_class = v.try(:evm_object_class)`.  Then the value can be checked in each leg of the `if` statement or even replace it with a `case/when` construct.

Reply:
Yeah I had this as a case/when but the final `elsif` had to become an `else\n  if`, was just trying to minimize changes in code that I don't understand well and we know isn't covered by specs

Reply:
right, so would it make sense to call `v.try(:evm_object_class)` once before the complex if/elsif/else statement?

Reply:
sure